js_modules/dagit/packages/core/src/gantt/__stories__/GanttChart.stories.tsx

```diff
@@ -1,60 +1,67 @@
-import {CustomTooltipProvider} from '@dagster-io/ui';
+import {MockedProvider, MockedResponse} from '@apollo/client/testing';
+import {Box, Button, Colors, CustomTooltipProvider} from '@dagster-io/ui';
 import {Meta} from '@storybook/react';
 import React, {useState} from 'react';
 
-import {RunStatus} from '../../graphql/types';
+import {RunStatus, buildRun, buildRunGroup, buildRunStatsSnapshot} from '../../graphql/types';
 import {extractMetadataFromLogs} from '../../runs/RunMetadataProvider';
 import {RunMetadataProviderMessageFragment} from '../../runs/types/RunMetadataProvider.types';
-import {StorybookProvider} from '../../testing/StorybookProvider';
+import {WorkspaceProvider} from '../../workspace/WorkspaceContext';
 import {IGanttNode} from '../Constants';
 import {GanttChart, GanttChartLoadingState} from '../GanttChart';
+import {RUN_GROUP_PANEL_QUERY} from '../RunGroupPanel';
+import {RunGroupPanelQuery, RunGroupPanelQueryVariables} from '../types/RunGroupPanel.types';
 
 const R1_START = 1619468000;
 const R2_START = 1619468000 + 30000;
 
-const APOLLO_MOCKS = {
-  RunGroupOrError: () => ({
-    __typename: 'RunGroup',
-    rootRunId: 'r1',
-    runs: [
-      {
-        __typename: 'Run',
-        id: 'r1',
-        runId: 'r1',
-        parentRunId: null,
-        status: RunStatus.FAILURE,
-        stepKeysToExecute: [],
-        pipelineName: 'Test',
-        tags: [],
-        stats: {
-          __typename: 'RunStatsSnapshot',
-          id: 'r1',
-          enqueuedTime: R1_START,
-          launchTime: R1_START + 12,
-          startTime: R1_START + 24,
-          endTime: R1_START + 400,
-        },
-      },
-      {
-        __typename: 'Run',
-        id: 'r2',
-        runId: 'r2',
-        parentRunId: 'r1',
-        status: RunStatus.STARTING,
-        stepKeysToExecute: [],
-        pipelineName: 'Test',
-        tags: [],
-        stats: {
-          __typename: 'RunStatsSnapshot',
-          id: 'r2',
-          enqueuedTime: R2_START,
-          launchTime: R2_START + 12,
-          startTime: null,
-          endTime: null,
-        },
-      },
-    ],
-  }),
+const runGroupMock: MockedResponse<RunGroupPanelQuery, RunGroupPanelQueryVariables> = {
```

Comment:
I actually added a new hook for this in internal:
https://github.com/dagster-io/internal/blob/356395a1617046d164bb434982bfaf7f64a73089/dagster-cloud/js_modules/dagit-cloud/src/testing/mocking.ts#L5

Reply:
so you could rewrite this as:

```
const runGroupMock: MockedResponse<RunGroupPanelQuery, RunGroupPanelQueryVariables> = buildQueryMock({
  query: RUN_GROUP_PANEL_QUERY,
  variables: {runId: 'r2'},
  data: {
     runGroupOrError: buildRunGroup({
        rootRunId: 'r1',
        runs: [
          buildRun({
            id: 'r1',
            runId: 'r1',
            parentRunId: null,
            status: RunStatus.FAILURE,
            stepKeysToExecute: [],
            pipelineName: 'Test',
            tags: [],
            stats: buildRunStatsSnapshot({
              id: 'r1',
              enqueuedTime: R1_START,
              launchTime: R1_START + 12,
              startTime: R1_START + 24,
              endTime: R1_START + 400,
            }),
          }),
          buildRun({
            id: 'r2',
            runId: 'r2',
            parentRunId: 'r1',
            status: RunStatus.STARTING,
            stepKeysToExecute: [],
            pipelineName: 'Test',
            tags: [],
            stats: buildRunStatsSnapshot({
              id: 'r2',
              enqueuedTime: R2_START,
              launchTime: R2_START + 12,
              startTime: null,
              endTime: null,
            }),
          }),
   }
});
```

Eliminates a little bit of boilerplate

Reply:
Oh nice! It looks like this is still in a automaterialize policy page file in OSS, I'll make a follow-up PR to adopt usage of it in OSS